packages/aws-cdk/lib/cli.ts

```diff
@@ -343,7 +372,7 @@ async function initCommandLine() {
       throw new Error('You must either specify a list of Stacks or the `--all` argument');
     }
 
-    args.STACKS = args.STACKS || [];
+    args.STACKS = args.STACKS ?? (args.STACK ? [args.STACK] : []);
     args.ENVIRONMENTS = args.ENVIRONMENTS || [];
```

Comment:
(Optional) Maybe for consistency here:

```suggestion
    args.ENVIRONMENTS = args.ENVIRONMENTS ?? [];
```